test/suites/standard/cleanup-data.robot

```diff
@@ -0,0 +1,158 @@
+    Verify Crio Resource Count    ${TRUE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    [Teardown]
+    ...    Start MicroShift And Wait Until Ready
+
+Verify Partial Cleanup Data
+    [Documentation]    Verify partial data clean scenarios
+
+    ${rc}=    Run MicroShift Cleanup Data    --all    --keep-images
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    ${rc}=    Run MicroShift Cleanup Data    --keep-images    --all
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${TRUE}
+
+    Start MicroShift And Wait Until Ready
+
+    ${rc}=    Run MicroShift Cleanup Data    --ovn
+    Should Be Equal As Integers    ${rc}    0
+    Verify Crio Resource Count    ${FALSE}
+    Verify MicroShift Data Directory And Processes    ${FALSE}
+
+    [Teardown]
+    ...    Start MicroShift And Wait Until Ready
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Start MicroShift And Wait Until Ready
+
+Start MicroShift And Wait Until Ready
+    [Documentation]    Start the service and wait until full initialized
+    Systemctl    enable    --now microshift
+    Restart Greenboot And Wait For Success
+
+Run MicroShift Cleanup Data
+    [Documentation]    Run the microshift-cleanup-data script and
+    ...    return its exit code
+    [Arguments]    ${cmd}    ${opt}=${EMPTY}
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    echo 1 | sudo microshift-cleanup-data ${cmd} ${opt}
+    ...    return_stdout=True    return_stderr=True    return_rc=True
+    RETURN    ${rc}
+
+Verify Crio Resource Count
+    [Documentation]    Make sure that cri-o pod, container
+    ...    and image counts are correct
+    [Arguments]    ${after_image_clean}
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    crictl pods | wc -l
+    ...    sudo=True    return_stdout=True    return_stderr=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+    Should Be Equal As Integers    ${stdout}    1
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    crictl ps | wc -l
+    ...    sudo=True    return_stdout=True    return_stderr=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+    Should Be Equal As Integers    ${stdout}    1
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ...    crictl images | wc -l
+    ...    sudo=True    return_stdout=True    return_stderr=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+    IF    ${after_image_clean}
+        Should Be Equal As Integers    ${stdout}    1
+    ELSE
+        Should Be True    ${stdout} > 1
+    END
+
+Verify MicroShift Data Directory And Processes
```

Comment:
Similarly here

- MicroShift Data Directory Should Exist
- MicroShift Data Directory Should Not Exist
- MicroShift Should Be Stopped
- MicroShift-etcd Should Be Stopped

etc.